locksmith/src/middlewares/signatureValidationMiddleware.ts

```diff
@@ -15,56 +16,78 @@ namespace SignatureValidationMiddleware {
     }
   }
 
-  const validatePayload = (payload: any, signee: string): Boolean => {
-    if (payload.lock) {
-      return (
-        ethJsUtil.toChecksumAddress(payload.lock.owner) ===
-        ethJsUtil.toChecksumAddress(signee)
-      )
-    } else {
-      return false
-    }
-  }
-  const validatePayloadContent = (payload: any): Boolean => {
-    if (payload.lock) {
-      if (payload.lock.name && payload.lock.owner && payload.lock.address) {
-        return true
-      }
-    }
-    return false
+  const validateSignee = (payload: any, signee: string): Boolean => {
+    return (
+      Normalizer.ethereumAddress(payload) === Normalizer.ethereumAddress(signee)
+    )
   }
 
-  export const process = (req: any, res: Response, next: any) => {
-    var signature = extractToken(req)
+  const validatePayloadContent = (
+    payload: any,
+    configuration: SignatureValidationConfiguration
+  ): Boolean => {
+    let result = configuration.required.every(element => {
+      return !(payload[element] == null)
+    })
 
-    if (signature === null) {
-      res.sendStatus(401)
-      return
-    }
-
-    let decodedSignature = Base64.decode(signature)
+    return result
+  }
 
+  const handleSignaturePresent = (
+    body: any,
+    signature: string,
+    configuration: SignatureValidationConfiguration
+  ) => {
     try {
       let signee = sigUtil.recoverTypedSignature({
-        data: req.body,
-        sig: decodedSignature,
+        data: body,
+        sig: signature,
       })
 
+      let potentialSignee: string =
+        body.message[configuration.name][configuration.signee]
+
       if (
-        req.body.message &&
-        validatePayload(req.body.message, signee) &&
-        validatePayloadContent(req.body.message)
+        body.message &&
+        validateSignee(potentialSignee, signee) &&
+        validatePayloadContent(body.message[configuration.name], configuration)
       ) {
-        req.owner = ethJsUtil.toChecksumAddress(signee)
-        next()
+        return Normalizer.ethereumAddress(signee)
       } else {
+        return null
+      }
+    } catch (e) {
+      return null
+    }
+  }
+
+  export const generateProcessor = (
```

Comment:
Could you add a JSDoc comment above to indicate the purpose of that function (generate signatures based on the payload)